src/ITensorsGPU.jl

```diff
@@ -60,4 +60,8 @@ export cu,
        productCuMPS,
        randomCuMPO,
        cuMPO
+
+function __init__()
+    ITensors.enable_combine_contract()
```

Comment:
This is kind of unfortunate since it will turn it on for non-GPU block sparse contractions as well.

The combine-contract logic [here](https://github.com/ITensor/ITensors.jl/blob/0784a2901694077fc27fd90db67b5c8c4c1b2004/src/itensor.jl#L1473-L1490) was a bad hack which should have been done within NDTensors but I mostly just added it as a quick test of that contraction strategy for block sparse tensors.

I'm trying to think of a simpler way of getting the functionality you want here but without going through that logic. Ultimately it kind of goes back to the point I was bringing up in [this issue](https://github.com/ITensor/ITensorsGPU.jl/issues/10) where in different cases you might want to overload contraction at different points. This case is basically saying that sometimes you want to overload the highest level `contract!!` directly when `cuTENSOR` is working, but sometimes you want it to pass through the transpose-transpose-gemm-transpose logic and do `gemm!`-like call. I noticed you already had some logic like that [here](https://github.com/ITensor/ITensorsGPU.jl/blob/3d6ff29d2737597be1caa57f5a812a5123b8a163/src/tensor/dense.jl#L195-L265). Ideally that could become a generic function where the only overload point is the `gemm!` call, but for now can you use that function in this case (calling whatever `gemm!` is appropriate)?

Reply:
Yeah, I could try that too.

Reply:
Although the `combine_contract` is convenient because it handles the combiner tensors for me :p

Reply:
Or perhaps in this case we hope that `cuTENSOR` fixes this limit and this is only a temporary workaround...

Reply:
Yeah, I'd prefer to keep it like this for now (as ugly as it is) and hope CUTENSOR fixes this soon.

Reply:
Right, I think the best way is to not go through the combiner at all and instead use [this](https://github.com/ITensor/ITensorsGPU.jl/blob/3d6ff29d2737597be1caa57f5a812a5123b8a163/src/tensor/dense.jl#L195-L265). Then you don't have to use a combiner at all. Could you try that? To be honest I don't think it is a good idea to use `enable_combine_contract` since then just loading this package changes the behavior (at least the performance) of other ITensor code, which could be very confusing for someone.

Reply:
Yeah, I'll give it a spin